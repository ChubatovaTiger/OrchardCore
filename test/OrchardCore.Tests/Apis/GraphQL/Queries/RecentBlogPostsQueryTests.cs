using System.Threading.Tasks;
using Assent;
using OrchardCore.Tests.Apis.Context;
using Xunit;

namespace OrchardCore.Tests.Apis.GraphQL.Queries
{
    public class RecentBlogPostsQueryTests
    {
        [Fact]
        public async Task ShouldListBlogPostWhenCallingAQuery()
        {
            await BlogContext.InitializeBlogAsync();

            var blogPostContentItemId = await BlogContext
                .GraphQLClient
                .Content
                .Create("BlogPost", builder =>
                {
                    builder
                        .WithField("Published", true);
                    builder
                        .WithField("Latest", true);

                    builder
                        .WithContentPart("TitlePart")
                        .AddField("Title", "Some sorta blogpost in a Query!");

                    builder
                        .WithContentPart("ContainedPart")
                        .AddField("ListContentItemId", BlogContext.BlogContentItemId);
                });

            var result = await BlogContext
                .GraphQLClient
                .Content
                .Query("RecentBlogPosts", builder =>
                {
                    builder
                        .WithNestedField("TitlePart")
                        .AddField("Title");
                });

            this.Assent(result.ToString());
        }
    }
}
