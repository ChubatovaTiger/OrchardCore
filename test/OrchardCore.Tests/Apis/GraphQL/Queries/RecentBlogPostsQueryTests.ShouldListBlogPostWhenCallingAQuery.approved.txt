{
  "data": {
    "recentBlogPosts": [
      {
        "titlePart": {
          "title": "Some sorta blogpost in a Query!"
        }
      },
      {
        "titlePart": {
          "title": "Man must explore, and this is exploration at its greatest"
        }
      }
    ]
  }
}